app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.java

```diff
@@ -384,50 +277,43 @@ public boolean onCreateOptionsMenu(Menu menu) {
     public boolean onPrepareOptionsMenu(Menu menu) {
         // Grab menu items for updating later
         switchCameraMenuItem = menu.findItem(R.id.switch_camera_menu_item);
+        screenCaptureMenuItem = menu.findItem(R.id.share_screen_menu_item);
         pauseVideoMenuItem = menu.findItem(R.id.pause_video_menu_item);
         pauseAudioMenuItem = menu.findItem(R.id.pause_audio_menu_item);
-        screenCaptureMenuItem = menu.findItem(R.id.share_screen_menu_item);
 
         // Screen sharing only available on lollipop and up
         screenCaptureMenuItem.setVisible(Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP);
 
+        roomManager.getViewState().observe(this, this::bindViewState);
```

Comment:
This is an odd place to observe the LiveData. Why was this moved from onCreate?

Reply:
Yeah I don't like it here either, but when binding the menu items for audio & video toggling, a NPE will get thrown since onCreateOptionsMenu gets invoked after onCreate.

Reply:
Another solution to this is to add another livedata observer function just for menu items. Might be a bit better, but downside is the view state will be bound in two functions which reduces readability of the class.